icaruscode/Decode/ChannelMapping/ChannelMapSQLite_tool.cc

```diff
@@ -101,11 +134,51 @@ ChannelMapSQLite::ChannelMapSQLite(fhicl::ParameterSet const &pset)
 }
 
 //------------------------------------------------------------------------------------------------------------------------------------------
+std::array<ChannelMapSQLite::TableNames_t, RunPeriods::NPeriods> const
+ChannelMapSQLite::TableNameSets{
+  TableNames_t{
+      "readout_boards"  // TPCfragmentMap
+    , "daq_channels"    // TPCreadoutBoardMap
+    , "pmt_placements"  // PMTfragmentMap
+    , "feb_channels"    // CRTsideMap
+    , "crtfeb"          // CRTtopMap
+  },
+  TableNames_t{
+      "readout_boards"          // TPCfragmentMap
+    , "daq_channels"            // TPCreadoutBoardMap
+    , "pmt_placements_Aug2023"  // PMTfragmentMap
+    , "feb_channels"            // CRTsideMap
+    , "crtfeb"                  // CRTtopMap
+  }
+};
 
-ChannelMapSQLite::~ChannelMapSQLite()
-{
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+bool ChannelMapSQLite::SelectPeriod(RunPeriod period) {
+  
+  auto const iPeriod = static_cast<unsigned int>(period);
+  TableNames_t const* newSet = &(TableNameSets.at(iPeriod));
+  
+  if (fCurrentTable == newSet) {
+    mf::LogDebug("ChannelMapSQLite")
+      << "Period #" << iPeriod << " already selected";
+  }
+  else if (fCurrentTable) {
+    mf::LogDebug("ChannelMapSQLite") << "Switched from period #"
+      << (fCurrentTable - &(TableNameSets[0])) << " to #" << iPeriod;
+  }
+  else {
+    mf::LogDebug("ChannelMapSQLite") << "Switching to period #" << iPeriod;
+  }
+  
+  if (fCurrentTable == newSet) return false;
```

Comment:
Is there a reason for not adding `return false` directly in the same `if` statement above? Just curious.

Reply:
Not a string one. I was separating the functional part (`if (fCurrentTable == newSet) return false;`) from he debugging message part, which may change or be removed (ok, more honestly, which was added later).